mithril-signer/src/services/signature_publisher/interface.rs

```diff
@@ -0,0 +1,16 @@
+use async_trait::async_trait;
+use mithril_common::entities::{ProtocolMessage, SignedEntityType, SingleSignatures};
+use mithril_common::StdResult;
+
+/// Publishes computed single signatures to a third party.
+#[cfg_attr(test, mockall::automock)]
+#[async_trait]
+pub trait SignaturePublisher: Send + Sync {
+    /// Publish computed single signatures.
+    async fn publish(
+        &self,
+        signed_entity_type: &SignedEntityType,
+        signatures: &SingleSignatures,
```

Comment:
```suggestion
        signature: &SingleSignatures,
```

Maybe it's also worth renaming `SingleSignatures` to `SingleSignature`

Reply:
I have also modified the HTTP implementation attribute name it http.rs